### ISSUE TITLE
MS_MUTABLE_ARRAY not detected for public static final array

### ISSUE THREAD

**Issue by csrma** [spotbugs/spotbugs#3465]
In the following code, the `static final` variable references an array which can be accessed and modified. However, this is not detected by SpotBugs for `MS_MUTABLE_ARRAY`.
```
class MutableArrayBug {
    // BUG: MS: Field is a mutable array (MS_MUTABLE_ARRAY)
    public static final int[] ARRAY = {1, 2, 3, 4, 5};

    public static void showBug() {
        System.out.println(ARRAY[0]);
        ARRAY[0] = 10;
        System.out.println(ARRAY[0]);
    }
}
```
SpotBugs version: `4.9.2`
Java: `Temurin openjdk 21.0.7`
Ran from CLI as the following two commands:
```
export SPOTBUGS_JAR=/path/to/spotbugs.jar
```
```
javac MutableArrayBug.java && \
java -jar $SPOTBUGS_JAR \
    -low \
    -effort:max \
    MutableArrayBug.class
```

**Comment by mek1986** [comment]
ARRAY references cannot be modified, which does not mean that the elements in the ARRAY array cannot be modified.

**Comment by csrma** [comment]
@mek1986 The [rule description](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#ms-field-is-a-mutable-array-ms-mutable-array) specifically mentions "modify the **contents** of the array".

Also, since it is a `final` variable, if we try to change its reference (e.g. assign a new array by `new int[]{6, 7, 8, 9, 10}`), that should give a compilation error anyway, so a spotbugs rule would not be required in that case. Of course unless I'm wrong or misunderstood the rule.
